x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerUtils.java

```diff
@@ -72,6 +81,35 @@ public static Tuple<PhysicalPlan, PhysicalPlan> breakPlanBetweenCoordinatorAndDa
         return new Tuple<>(coordinatorPlan, dataNodePlan.get());
     }
 
+    public static PhysicalPlan dataNodeReductionPlan(LogicalPlan plan, PhysicalPlan unused) {
+        var pipelineBreakers = plan.collectFirstChildren(Mapper::isPipelineBreaker);
+
+        if (pipelineBreakers.isEmpty() == false) {
+            UnaryPlan pipelineBreaker = (UnaryPlan) pipelineBreakers.get(0);
+            if (pipelineBreaker instanceof TopN topN) {
+                return new TopNExec(topN.source(), unused, topN.order(), topN.limit(), 2000);
```

Comment:
I guess you were planning to add a comment for this 2K, but leaving a note as a reminder.

Reply:
So far, I couldn't find a solid solution to have a calculated value for this. Used 2000 for now with a follow up task to dig deeper into getting a more accurate value.